experimental/language-generation/csharp_dotnetcore/05.multi-turn-prompt/Dialogs/UserProfileDialog.cs

```diff
@@ -64,20 +63,20 @@ private static async Task<DialogTurnResult> NameStepAsync(WaterfallStepContext s
         {
             stepContext.Values["transport"] = ((FoundChoice)stepContext.Result).Value;
 
-            return await stepContext.PromptAsync(nameof(TextPrompt), new PromptOptions { Prompt = ActivityFactory.CreateActivity(_lgEngine.EvaluateTemplate("AskForName").ToString()) }, cancellationToken);
+            return await stepContext.PromptAsync(nameof(TextPrompt), new PromptOptions { Prompt = ActivityFactory.FromObject(_lgTemplates.Evaluate("AskForName")) }, cancellationToken);
         }
 
         private async Task<DialogTurnResult> NameConfirmStepAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
         {
             stepContext.Values["name"] = (string)stepContext.Result;
 
             // We can send messages to the user at any point in the WaterfallStep.
-            await stepContext.Context.SendActivityAsync(ActivityFactory.CreateActivity(_lgEngine.EvaluateTemplate("AckName", new {
+            await stepContext.Context.SendActivityAsync(ActivityFactory.FromObject(_lgTemplates.Evaluate("AckName", new {
```

Comment:
>FromObject(_lgTemplates.Evaluate [](start = 72, length = 32)

Makes me wonder if we should have an extension helper method:
ActivityFactory.Evaluate(templates, "AckName", ... );

Reply:
Sure. can track it as a backlog ticket.